File: backend/LookApp.Database/Models/Record.cs
using System;

namespace LookApp.Database.Models
{
    public class Record
    {
        public int Id { get; set; }
        public DateTime Date { get; set; }
        public string Note { get; set; }
        public double Value { get; set; }

        public int CategoryId { get; set; }
        public Category Category { get; set; }
    }
}

File: frontend/dashboard/src/app/shared/models/record.ts
export class RecordModel{
  id?: number;
  date: Date;
  value: number;
  categoryId: number;
  note?: string;

  public constructor(date: Date, value: number, categoryId: number, note: string) {
    this.date = date;
    this.value = value;
    this.categoryId = categoryId;
    this.note = note;
  }
}
